packages/jest-cli/src/jest.js

```diff
@@ -290,175 +48,40 @@ const runCLI = (
 
   readConfig(argv, root)
     .then(config => {
-      if (argv.debug) {
-        /* $FlowFixMe */
-        const testFramework = require(config.testRunner);
-        pipe.write('jest version = ' + VERSION + '\n');
-        pipe.write('test framework = ' + testFramework.name + '\n');
-        pipe.write('config = ' + JSON.stringify(config, null, '  ') + '\n');
-      }
-      if (argv.watch || argv.watchAll) {
-        setWatchMode(argv, argv.watch ? 'watch' : 'watchAll', {
-          pattern: argv._,
-        });
-
-        let currentPattern = '';
-        let hasSnapshotFailure = false;
-        let isEnteringPattern = false;
-        let isRunning = false;
-        let testWatcher;
-        let timer;
-        let displayHelp = true;
-
-        const writeCurrentPattern = () => {
-          clearLine(pipe);
-          pipe.write(chalk.dim(' pattern \u203A ') + currentPattern);
-        };
-
-        const startRun = (overrideConfig: Object = {}) => {
-          if (isRunning) {
-            return null;
-          }
+      createDirectory(config.cacheDirectory);
+      const jestHasteMap = Runtime.createHasteMap(config, {
+        console: new Console(pipe, pipe),
+        maxWorkers: getMaxWorkers(argv),
+        resetCache: !config.cache,
+        watch: config.watch,
+      });
 
-          testWatcher = new TestWatcher({isWatchMode: true});
-          pipe.write(CLEAR);
+      return jestHasteMap.build().then(
+        hasteMap => ({
+          hasteFS: hasteMap.hasteFS,
+          resolver: Runtime.createResolver(config, hasteMap.moduleMap),
+        }),
+        error => {
+          throw error;
+        },
+      )
+      .then(hasteContext => {
+        if (argv.debug) {
```

Comment:
Can we move this before creating the haste map again? I don't think we need to create the haste map before logging this. Could also factor this if out into a `logDebugMessages`.